crates/primitives-traits/src/block/body.rs

```diff
@@ -35,6 +35,13 @@ pub trait BlockBody:
     /// Ommer header type.
     type OmmerHeader: BlockHeader;
 
+    /// Returns a new instance of the [`BlockBody`].
+    fn new(
+        transactions: &[Self::Transaction],
+        ommers: &[Self::OmmerHeader],
+        withdrawals: Option<Withdrawals>,
+    ) -> Self;
```

Comment:
hmm, don't think we can do this because this now restricts the body to txs + ommers + withdrawals 

so what if the body has an additional field?

Reply:
hmmm, I do need some way to transform a block for prewarming, since once we're in the engine we have no concrete types. I have to take parts of one `Block` and use it to create another `Block` with only specific transactions. Maybe we can solve this with a different type of method?

Reply:
I guess we could also solve this with a filter method on body, that returns a new body

Reply:
ah, we can just have setters